oktapam/resource_cloud_connection.go

```diff
@@ -0,0 +1,180 @@
+package oktapam
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/okta/terraform-provider-oktapam/oktapam/client"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/descriptions"
+	"github.com/okta/terraform-provider-oktapam/oktapam/logging"
+	"github.com/okta/terraform-provider-oktapam/oktapam/utils"
+)
+
+func resourceCloudConnection() *schema.Resource {
+	return &schema.Resource{
+		Description:   descriptions.ResourceCloudConnection,
+		ReadContext:   resourceCloudConnectionRead,
+		CreateContext: resourceCloudConnectionCreate,
+		DeleteContext: resourceCloudConnectionDelete,
+		UpdateContext: resourceCloudConnectionUpdate,
+		Schema: map[string]*schema.Schema{
+			attributes.ID: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			attributes.Name: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.Name,
+			},
+			attributes.CloudConnectionProvider: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.CloudConnectionProvider,
+			},
+			attributes.CloudConnectionDetails: {
+				Type:        schema.TypeList,
+				MinItems:    1,
+				MaxItems:    1,
+				Required:    true,
+				ForceNew:    true,
+				Description: descriptions.CloudConnectionDetails,
+				Elem: &schema.Resource{
```

Comment:
So this is locking in the general structure of these fields.  If/when we do add a new cloud provider, it will need to work within this structure.  Moving to a different structure would essentially require deprecating this resource and creating a new one, which is not the best.

Reply:
Done